playbook.yml

```diff
@@ -0,0 +1,85 @@
+- name: Deploy app to webservers
+  hosts: webservers
+  remote_user: ec2-user
+  tasks:
+    - name: include vault
+      ansible.builtin.include_vars:
+        file: secrets.yml
+
+    - name: Create todoapp user
+      ansible.builtin.user:
+        name: todoapp
+      become: yes
+
+    - name: Install git
+      ansible.builtin.yum:
+        name: git
+        state: installed
+      become: yes
+
+    - name: Check python3 version
+      ansible.builtin.command:
+        cmd: python3 --version
+      register: python_version_output
+
+    - name: Copy python3 installation script
+      ansible.builtin.copy:
+        src: ~/module_4/install_python.sh
+        dest: ~/
+      when: "python_version_output['stdout'] != 'Python 3.10.2'"
```

Comment:
Nice work with this - it's a little more fiddly but this looks like a clean solution 👍

Reply:
My project didn't like the version of Python3 installed by yum so I had to install 3.10. I thought creating a shell script based on the FAQ would be the best option rather than running each command in the playbook.